rust/worker/src/execution/orchestration/compact.rs

```diff
@@ -653,7 +656,38 @@ impl Handler<TaskResult<MaterializeLogOutput, MaterializeLogOperatorError>>
             }
         };
 
-        self.write(materialized_result, ctx.receiver(), ctx).await;
+        if materialized_result.get_materialized_records().is_empty() {
+            // There is no delta after materialization, so there is no need to write
+            self.num_write_tasks -= 1;
+            if self.num_write_tasks == 0 {
+                if self.dirty {
+                    let writer_res = self.get_segment_writers().await;
```

Comment:
I ended up not storing writers on the struct in my base PR, so I think this may be buggy